src/query/functions/binary/or.go

```diff
@@ -61,11 +70,28 @@ func makeOrBlock(
 		return nil, err
 	}
 
-	index := 0
-	for ; lIter.Next(); index++ {
+	values := make([]float64, len(combinedSeriesMeta))
+	for index := 0; lIter.Next(); index++ {
+		if !rIter.Next() {
+			return nil, errRExhausted
+		}
+
 		lStep := lIter.Current()
 		lValues := lStep.Values()
-		if err := builder.AppendValues(index, lValues); err != nil {
+		copy(values, lValues)
+		rStep := rIter.Current()
+		rValues := rStep.Values()
+		for iterIndex, valueIndex := range indices {
+			if valueIndex >= len(lValues) {
+				values[valueIndex] = rValues[iterIndex]
+			} else {
+				if math.IsNaN(values[valueIndex]) {
```

Comment:
nit: Perhaps move this into an `else if { ... }` to reduce the next level of nesting?